lib/pmdtester/parsers/options.rb

```diff
@@ -0,0 +1,53 @@
+require 'slop'
+
+module PmdTester
+  class Options
+
+    attr_reader :local_git_repo
+    attr_reader :base_branch
+    attr_reader :patch_branch
+    attr_reader :base_config
+    attr_reader :patch_config
+    attr_reader :config
+    attr_reader :project_list
+    attr_reader :mode
+
+    def initialize(argv)
+      options = parse(argv)
+      @local_git_repo = options[:LocalGitRepo]
+      @base_branch = options[:baseBranch]
+      @patch_branch = options[:patchBranch]
+      @base_config = options[:baseConfig]
+      @patch_config = options[:patchConfig]
+      @config = options[:config]
+      @project_list = options[:listOfProjects]
+      @mode = options[:mode]
+    end
+
+    private
+
+    def parse(argv)
+      Slop.parse argv do |o|
+        o.string '-r', '--LocalGitRepo', 'path to the local PMD repository'
+        o.string '-b', '--baseBranch', 'name of the base branch in local PMD repository'
+        o.string '-p', '--patchBranch', 'name of the patch branch in local PMD repository'
+        o.string '-bc', '--baseConfig', 'path to the base PMD configuration file'
+        o.string '-pc', '--patchConfig', 'path to the patch PMD configuration file'
+        o.string '-c', '--config', 'path to the base and patch PMD configuration file'
+        o.string '-l', '--listOfProjects', 'path to the file which contains the list of standard projects'
```

Comment:
I'd change the long option names to something more standard in other command line tools: all lowercase, split with dash when needed. ie: --local-git-repo or --patch-branch

Reply:
Done.